[docs] CDCSDK: Add documentation for transaction ordering with CDC (#18551)
yugabyte/yugabyte-db: docs/content/preview/explore/change-data-capture/debezium-connector-yugabytedb.md

| :--- | :--- |
| transaction.ordering | Whether to enable ordering of transactions by their commit time. |
| transforms | Logical name of the transformer being used. In our case, we will be using `Reroute` for example purposes. |
| transforms.Reroute.topic.regex | Specifies a regular expression that the transformation applies to each change event record to determine if it should be routed to a particular topic. |
| transforms.Reroute.topic.replacement | A regular expression that represents the destination topic name. |
| transforms.Reroute.type | Transformer class to be used. |
| transforms.Reroute.key.field.regex | Specifies a regular expression that the transformation applies to the default destination topic name to capture one or more groups of characters. |
| transforms.Reroute.key.field.replacement | Specifies a regular expression for determining the value of the inserted key field in terms of those captured groups. |
| provide.transaction.metadata | Whether to generate events with transaction boundaries. |

For usage example, refer to our [example repository](https://github.com/yugabyte/cdc-examples/tree/main/consistent-streaming).

### Transaction boundaries

> The connector publishes metadata which can be used to distinguish transaction boundaries for a downstream application to implement atomicity. Once the configuration property `provide.transaction.metadata` (see [transaction metadata]()) is enabled, the connector will also publish events indicating the beginning and end of the transaction.

Reviewer: ```suggestion
The connector publishes metadata that can be used to distinguish transaction boundaries for a downstream application to implement atomicity. Once the configuration property `provide.transaction.metadata` is enabled, the connector will also publish events indicating the beginning and end of the transaction. For more information, see [Transaction metadata](#transaction-metadata).
```

### Prerequisites

> * Stream ID should be created in the `EXPLICIT` checkpointing mode. To know more, see [yb-admin create\_change\_data_stream](../../admin/yb-admin#create_change_data_stream).

Reviewer: ```suggestion
* Create the Stream ID should in the `EXPLICIT` checkpointing mode. For more information, see [yb-admin create\_change\_data_stream](../../../admin/yb-admin#create_change_data_stream).
```

> * The connector should always be run with a single task i.e. `tasks.max` should always be set to 1.

Reviewer: ```suggestion
* You should always run the connector with a single task, that is, `tasks.max` should always be set to 1.
```

> * Create the Stream ID should in the `EXPLICIT` checkpointing mode. For more information, see [yb-admin create\_change\_data_stream](../../../admin/yb-admin#create_change_data_stream).

Reviewer: ```suggestion
* Create the Stream ID should in the `EXPLICIT` checkpointing mode. For more information, see [yb-admin create\_change\_data_stream](../../../admin/yb-admin#create-change-data-stream).
```

> ### Known limitations with transaction ordering

Reviewer: ```suggestion
### Known limitations
```

> * Transactional ordering is currently not supported with schema evolution. Issue [18476](https://github.com/yugabyte/yugabyte-db/issues/18476).

Reviewer: ```suggestion
* Transactional ordering is currently not supported with schema evolution. See issue [18476](https://github.com/yugabyte/yugabyte-db/issues/18476).
```
